File: exercise/python/day07/production/workshop.py
from enum import Enum
from typing import Optional


class Status(Enum):
    PRODUCING = "Producing"
    PRODUCED = "Produced"


class Gift:
    def __init__(self, name: str, status: Status = Status.PRODUCING):
        self.name = name
        self.status = status


class Workshop:
    def __init__(self):
        self.gifts = []

    def add_gift(self, gift: Gift):
        self.gifts.append(gift)

    def complete_gift(self, name: str) -> Optional[Gift]:
        for gift in self.gifts:
            if gift.name == name:
                gift.status = Status.PRODUCED
                return gift
        return None

File: exercise/scala/day07/src/main/scala/production/Workshop.scala
package production

import production.Status.*

enum Status {
  case Producing, Produced
}

case class Gift(name: String, var status: Status = Status.Producing)

class Workshop {
  private val gifts = scala.collection.mutable.ListBuffer[Gift]()

  def addGift(gift: Gift): Unit = {
    gifts += gift
  }

  def completeGift(name: String): Option[Gift] = {
    gifts.find(_.name == name).map { gift =>
      gift.copy(status = Produced)
    }
  }
}